### DISCUSSION TITLE
Web API Patch returns 404

### DISCUSSION THREAD

**Discussion by mrpn** [pocketbase/pocketbase#4964] [Q&A] [answered]
Hello,

I’m encountering a strange bug. When I make a PATCH request to update a record, I receive a 404 error. Despite this error, the record is successfully updated.

Here are the details:
	•	I have ruled out API filter issues; the update request is set to be accessible by everyone.
	•	The record is part of a relationship and is used in another table.
	•	There are no field rules or indexes set.
	•	I have set custom IDs like 000000000000010, which might cause an issue, but a new table with the same logic works without problems.
	•	There are no hooks tied to this table, so we can rule that out.
	•	I am using the latest 22.12 version.

I have tried everything I can think of but I’m stuck. How can I debug or find the reason for getting a 404 error for my update request even though the data is updated?

Thank you

HTTP response
```json
{
  "code": 404,
  "message": "The requested resource wasn't found.",
  "data": {}
}
```

PB Log 
```json
{
  "id": "6kjq6xaogdojjpp",
  "created": "2024-05-20 14:06:29.859Z",
  "updated": "2024-05-20 14:06:29.859Z",
  "data": {
    "auth": "guest",
    "details": "sql: no rows in result set",
    "error": "The requested resource wasn't found.",
    "execTime": 2.27089,
    "method": "PATCH",
    "referer": "",
    "remoteIp": "192.168.0.2",
    "status": 404,
    "type": "request",
    "url": "/api/collections/XXX/records/000000000000010",
    "userAgent": "python-requests/2.31.0",
    "userIp": "xx.xx.xx.xx"
  },
  "message": "PATCH /api/collections/XXX/records/000000000000010",
  "level": 8
}
```

Collection details 
```
{
  "id": "k3ruqi49fpb81r8",
  "created": "2024-01-07 02:01:51.215Z",
  "updated": "2024-05-20 15:02:39.802Z",
  "name": "testbase",
  "type": "base",
  "system": false,
  "schema": [
    {
      "system": false,
      "id": "kpqanmfb",
      "name": "name",
      "type": "select",
      "required": false,
      "presentable": true,
      "unique": false,
      "options": {
        "maxSelect": 1,
        "values": [
          "bb",
          "bbb"
        ]
      }
    },
    {
      "system": false,
      "id": "mjkvtdot",
      "name": "aaaa",
      "type": "select",
      "required": false,
      "presentable": true,
      "unique": false,
      "options": {
        "maxSelect": 1,
        "values": [
          "aa",
          "aaa"
        ]
      }
    },
    {
      "system": false,
      "id": "4hcoquro",
      "name": "xxxx",
      "type": "select",
      "required": false,
      "presentable": true,
      "unique": false,
      "options": {
        "maxSelect": 1,
        "values": [
          "xx",
          "xxx"
        ]
      }
    },
    {
      "system": false,
      "id": "1ja9zrsk",
      "name": "abcdate",
      "type": "date",
      "required": false,
      "presentable": false,
      "unique": false,
      "options": {
        "min": "",
        "max": ""
      }
    },
    {
      "system": false,
      "id": "jlfvb9pm",
      "name": "abc2date",
      "type": "date",
      "required": false,
      "presentable": false,
      "unique": false,
      "options": {
        "min": "",
        "max": ""
      }
    },
    {
      "system": false,
      "id": "5s2kp9qo",
      "name": "emax",
      "type": "date",
      "required": false,
      "presentable": false,
      "unique": false,
      "options": {
        "min": "",
        "max": ""
      }
    },
    {
      "system": false,
      "id": "29n1sqgs",
      "name": "percentage",
      "type": "number",
      "required": false,
      "presentable": false,
      "unique": false,
      "options": {
        "min": 0,
        "max": 100,
        "noDecimal": true
      }
    },
    {
      "system": false,
      "id": "4xjfpshk",
      "name": "abc",
      "type": "text",
      "required": false,
      "presentable": false,
      "unique": false,
      "options": {
        "min": null,
        "max": null,
        "pattern": ""
      }
    },
    {
      "system": false,
      "id": "swqz8wce",
      "name": "abc2",
      "type": "text",
      "required": false,
      "presentable": false,
      "unique": false,
      "options": {
        "min": null,
        "max": null,
        "pattern": ""
      }
    }
  ],
  "indexes": [],
  "listRule": "",
  "viewRule": "",
  "createRule": "",
  "updateRule": "",
  "deleteRule": "",
  "options": {}
}
```

**Comment by ganigeorgiev** [comment]
I'm not able to reproduce the issue with the provided collection.

Double check that you are sending the request to the correct collection (_I'm not sure if it is just the example but in the logs the collection is `XXX` while the provided collection is `testbase`_).

If you are able to update the record from the Admin UI then most likely there is something wrong with your PATCH request but without details how you are actually sending it there is no much I can help with.

**Reply by ganigeorgiev** [reply]
I forgot to mention that you can also start the executable with `--dev` and see if there is a more detailed log in the stderr.

**Reply by ganigeorgiev** [reply]
> I receive a 404 error. Despite this error, the record is successfully updated.

This doesn't really make sense.

Are you sure that the 404 error is from this request?

You've mentioned that _"There are no hooks tied to this table"_ but if you have an error in an `after*` hook then it is possible to receive an error response even if the related hook record was successfully persisted.

**Comment by mrpn** [comment, marked as answer]
@ganigeorgiev Yes, the request was from that collection. I redacted the collection name, which is why there was a discrepancy between XXX and testbase.

I managed to fix the issue by updating the index rule from the relate table/base.

My index rule looks like this  this:

```sql
CREATE UNIQUE INDEX `idx_m8K0jwS` ON `events` (
  `date`,
  `type`,
  `unknowns`,
  `testbase`
)
```

I toggled the UNIQUE index on and off, and suddenly it started working. I’m not sure why this happened or how it fixed the issue, but since the problem is resolved and not reproducible, I wanted to share this for the record.

Sorry for taking your time @ganigeorgiev
